components/metrics/component.go

```diff
@@ -0,0 +1,138 @@
+		Run:      run,
+		IsEnabled: func(container *dig.Container) bool {
+			if err := container.Provide(createCollector); err != nil {
+				panic(ierrors.Wrap(err, "failed to provide collector"))
+			}
+
+			return ParamsMetrics.Enabled
+		},
+	}
+}
+
+// PluginName is the name of the metrics collector plugin.
+var (
+	Component *app.Component
+	deps      dependencies
+
+	server *http.Server
+)
+
+type dependencies struct {
+	dig.In
+
+	Local    *peer.Local
+	Protocol *protocol.Protocol
+
+	Collector *collector.Collector
+}
+
+func run() error {
+	Component.LogInfo("Starting Prometheus exporter ...")
+
+	if ParamsMetrics.GoMetrics {
+		deps.Collector.Registry.MustRegister(collectors.NewGoCollector())
+	}
+	if ParamsMetrics.ProcessMetrics {
+		deps.Collector.Registry.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	}
+
+	registerMetrics()
+
+	return Component.Daemon().BackgroundWorker("Prometheus exporter", func(ctx context.Context) {
+		Component.LogInfo("Starting Prometheus exporter ... done")
+
+		engine := echo.New()
+		engine.Use(middleware.Recover())
+
+		engine.GET("/metrics", func(c echo.Context) error {
+			deps.Collector.Collect()
+
+			handler := promhttp.HandlerFor(
+				deps.Collector.Registry,
+				promhttp.HandlerOpts{
+					EnableOpenMetrics: true,
+				},
+			)
+			if ParamsMetrics.PromhttpMetrics {
+				handler = promhttp.InstrumentMetricHandler(deps.Collector.Registry, handler)
+			}
+			handler.ServeHTTP(c.Response().Writer, c.Request())
+
+			return nil
+		})
+		bindAddr := ParamsMetrics.BindAddress
+		server = &http.Server{Addr: bindAddr, Handler: engine, ReadTimeout: 5 * time.Second, WriteTimeout: 5 * time.Second}
+
+		go func() {
+			Component.LogInfof("You can now access the Prometheus exporter using: http://%s/metrics", bindAddr)
+			if err := server.ListenAndServe(); err != nil && !ierrors.Is(err, http.ErrServerClosed) {
+				Component.LogError("Stopping Prometheus exporter due to an error ... done")
+			}
+		}()
+
+		<-ctx.Done()
+		Component.LogInfo("Stopping Prometheus exporter ...")
+
+		shutdownCtx, shutdownCtxCancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer shutdownCtxCancel()
+
+		// shutdown collector to stop all pruning executors
+		deps.Collector.Shutdown()
+
+		//nolint:contextcheck // false positive
+		err := server.Shutdown(shutdownCtx)
+		if err != nil {
+			Component.LogWarn(err)
+		}
+
+		Component.LogInfo("Stopping Prometheus exporter ... done")
+	}, daemon.PriorityMetrics)
+}
+
+func createCollector() *collector.Collector {
+	return collector.New()
+}
+
+func registerMetrics() {
+	deps.Collector.RegisterCollection(TangleMetrics)
+	deps.Collector.RegisterCollection(ConflictMetrics)
+	deps.Collector.RegisterCollection(InfoMetrics)
+	deps.Collector.RegisterCollection(DBMetrics)
+	deps.Collector.RegisterCollection(CommitmentsMetrics)
+	deps.Collector.RegisterCollection(SlotMetrics)
+	deps.Collector.RegisterCollection(AccountMetrics)
```

Comment:
I personally wouldn't over-complicate this component, especially because it should disabled by default and a dedicated 1 gouroutine worker group is responsible to treat events asynchronously.